### PR TITLE
(PXP-10141): feat/studyViewerLoginAlert: add new text to study viewer login alert

### PR DESCRIPTION
**Pull request by ocshawn**
Jira Ticket: [PXP-10141](https://ctds-planx.atlassian.net/browse/PXP-10141)

### New Features
add new text to study viewer login alert

### CODE REVIEW

File: src/StudyViewer/StudyDetails.jsx [hunk 1]
```diff
@@ -298,6 +298,13 @@ class StudyDetails extends React.Component {
      // button. if there are more than 1 with different configs, TODO fix
      const requestAccessConfig = this.props.studyViewerConfig.buttons && this.props.studyViewerConfig.buttons.find((e) => e.type === 'request_access');
 
+     let loginAlertMessage = 'Researchers are required to log in to request access to this dataset. Use the button above to login or create an account.';
+     if (!(!this.props.data.requiredIdpField
+      || this.props.data.requiredIdpField === this.props.user?.idp)) {
```
Comment (mfshao) [inline src/StudyViewer/StudyDetails.jsx:303]: isn't this equivalent to  `if (this.props.data?.requiredIdpField && this.props.data.requiredIdpField === this.props.user?.idp)` ?

File: src/StudyViewer/StudyDetails.jsx [hunk 2]
```diff
@@ -298,6 +298,12 @@ class StudyDetails extends React.Component {
      // button. if there are more than 1 with different configs, TODO fix
      const requestAccessConfig = this.props.studyViewerConfig.buttons && this.props.studyViewerConfig.buttons.find((e) => e.type === 'request_access');
 
+     let loginAlertMessage = 'Researchers are required to log in to request access to this dataset. Use the button above to login or create an account.';
+     if (this.props.data.requiredIdpField && this.props.data.requiredIdpField === this.props.user?.idp)) {
```
Comment (ocshawn) [inline src/StudyViewer/StudyDetails.jsx:302]:
```suggestion
     if (this.props.data.requiredIdpField && this.props.data.requiredIdpField !== this.props.user?.idp)) {
```

File: src/StudyViewer/StudyDetails.jsx (hunk 1 above)
Comment (ocshawn) [reply src/StudyViewer/StudyDetails.jsx]: thanks for pointing out i think its the same as 
`this.props.data?.requiredIdpField && this.props.data.requiredIdpField !== this.props.user?.idp`

**Comment by mfshao** [comment]
I think there are some issues with those brackets but the change to the if logic lgtm

File: src/StudyViewer/StudyDetails.jsx [hunk 3]
```diff
@@ -298,6 +298,12 @@ class StudyDetails extends React.Component {
      // button. if there are more than 1 with different configs, TODO fix
      const requestAccessConfig = this.props.studyViewerConfig.buttons && this.props.studyViewerConfig.buttons.find((e) => e.type === 'request_access');
 
+     let loginAlertMessage = 'Researchers are required to log in to request access to this dataset. Use the button above to login or create an account.';
+     if (this.props.data.requiredIdpField && this.props.data.requiredIdpField !== this.props.user?.idp)) {
```
Comment (ocshawn) [inline src/StudyViewer/StudyDetails.jsx:302]:
```suggestion
     if (this.props.data.requiredIdpField && this.props.data.requiredIdpField !== this.props.user?.idp) {
```

**Review by mfshao — APPROVED** [review]

**Pull request merged**